src/api.rs

```diff
@@ -128,7 +146,32 @@ impl Client {
         }
     }
 
-    fn fetch_resource(url: &str, api_key: &str) -> Result<String, NewsApiError> {
+    async fn fetch_resource(url: &str, api_key: &str) -> Result<String, NewsApiError> {
+        static CLIENT_USER_AGENT: &str = concat!(
+            "rust-",
+            env!("CARGO_PKG_NAME"),
+            "/",
+            env!("CARGO_PKG_VERSION"),
+        );
```

Comment:
This duplicated bit of code can be factored out into a function

```
    fn create_user_agent() -> String {
      concat!(
          "rust-",
          env!("CARGO_PKG_NAME"),
          "/",
          env!("CARGO_PKG_VERSION"),
      ).to_owned()
    }
```

Reply:
This is an old code. For some reason, it was marked as changed. :thinking: 
But I can change this.

Reply:
This is some existing code. For some reason, it was marked as changed. thinking
But I can change this.